redis-hbo-provider/src/test/java/com/facebook/presto/statistic/TestRedisProviderPlugin.java

```diff
@@ -15,21 +15,37 @@
 
 import com.facebook.presto.spi.statistics.HistoryBasedPlanStatisticsProvider;
 import com.github.fppt.jedismock.RedisServer;
+import io.lettuce.core.RedisURI;
 import org.testng.annotations.Test;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import static com.facebook.airlift.testing.Assertions.assertInstanceOf;
 
 public class TestRedisProviderPlugin
 {
-    private final String redisHboPropertyFilePath = "src/test/java/resources/test-redis-hbo-provider.properties";
+    private Map<String, String> createConfigMap(String serverUri)
+    {
+        Map<String, String> configMap = new HashMap<>();
+        configMap.put("coordinator", "true");
+        configMap.put("hbo.redis-provider.enabled", "true");
+        configMap.put("hbo.redis-provider.total-fetch-timeoutms", "5000");
+        configMap.put("hbo.redis-provider.total-set-timeoutms", "5000");
+        configMap.put("hbo.redis-provider.default-ttl-seconds", "4320000");
+        configMap.put("hbo.redis-provider.cluster-mode-enabled", "false");
+        configMap.put("hbo.redis-provider.server_uri", serverUri);
+        return configMap;
+    }
 
     @Test
     public void testStartup()
             throws Exception
     {
-        RedisServer server = RedisServer.newRedisServer(57872);
+        RedisServer server = RedisServer.newRedisServer(); // bind to a random port
         server.start();
-        RedisProviderPlugin plugin = new RedisProviderPlugin(redisHboPropertyFilePath);
+        RedisProviderPlugin plugin = new RedisProviderPlugin(
+                createConfigMap(RedisURI.create(server.getHost(), server.getBindPort()).toString()));
```

Comment:
Removed the resources file because the plugin expects the resources file to contain the uri but since we are now generating the port randomly, we would not know beforehand. This was the easiest modification